common/src/main/java/com/google/auto/common/BasicAnnotationProcessor.java

```diff
@@ -106,18 +110,22 @@
  */
 public abstract class BasicAnnotationProcessor extends AbstractProcessor {
 
-  private final Set<ElementName> deferredElementNames = new LinkedHashSet<>();
-  private final SetMultimap<Step, ElementName> elementsDeferredBySteps =
+  /* For every element that is not module/package, to be well-formed its
+   * enclosing-type in its entirety should be well-formed. Since modules
+   * don't get annotated (and are not supported here) they can be ignored.
+   */
+  private final Set<PackageOrTypeElementFactory> deferredElementFactories = new LinkedHashSet<>();
```

Comment:
Another subjective matter regarding the naming: The first time I read the code, I was somewhat confused about why the code has both `deferredElement**` and `elementsDeferredBySteps`. That is why I personally prefer renaming the first to signify those are the elements defered due to being ill-formed.

Reply:
Yeah, we're still debating the names. Naming Is Hard.

Reply:
Another naming caught my eye now, and I just wanted to share my opinion again. :) I still prefer my *Blueprint* name idea, since to me *factory* usually means that something will be created. However, that is not the case here. The element is looked for (not created) that has certain characteristics.

Reply:
Naming Is Still Hard. :-) I think a Blueprint also implies creation, doesn't it? If it's something you use to find an existing thing, that's more like a Key. But also, it may actually be that a new object is created when you ask for the same method in a later round. And, in practice it's not unusual for a factory to be able to return an existing cached object rather than a new one.